### PR TITLE
Add a flag a logic to override k8scp compute api endpoint

### PR DESCRIPTION
**Pull request by panslava**
- Should not change any behaviour when flag is not set

/assign @swetharepakula @spencerhance 

This is mostly rollforward of https://github.com/kubernetes/ingress-gce/pull/2060 but with changing to override endpoint conditionally, and using flag as a value

### CODE REVIEW

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

File: cmd/glbc/main.go [hunk 1]
```diff
@@ -213,6 +214,16 @@ func main() {
 	kubeSystemUID := kubeSystemNS.GetUID()
 
 	cloud := app.NewGCEClient(rootLogger)
+
+	if flags.F.OverrideComputeAPIEndpoint != "" {
+		// Globally set the domain for all urls generated by GoogleCloudPlatform/k8s-cloud-provider.
+		// The cloud object is configured by the gce.conf file and parsed in app.NewGCEClient().
+		// basePath will be of the form <URL>/v1
+		domain := utils.GetDomainFromGABasePath(flags.F.OverrideComputeAPIEndpoint)
+		klog.Infof("Overriding k8s-cloud-provider API Domain to %q", domain)
```
Comment (alexkats) [inline cmd/glbc/main.go:223]: Please use context logger, `rootLogger` in this case

Comment (panslava) [reply cmd/glbc/main.go]: thank you, fixed!

**Pull request merged**